[Sng Haoren] iP

misterpuffin opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen (source)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
- All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is __FREE__!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

misterpuffin committed:
Implement Level 1 - Greet, Echo, Exit

misterpuffin committed:
Implemented Level 2 - Add, List

misterpuffin committed:
Implement Level 3 - Mark as Done

misterpuffin committed:
Implemented Level 4 - Todo, Event, Deadline

misterpuffin committed:
Add automated testing

misterpuffin committed:
Add error handling and refactor

misterpuffin committed:
Add delete command

misterpuffin committed:
Delete DukeHandler.java

misterpuffin committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

misterpuffin committed:
Implement Level 7 - Save into FileStorage

misterpuffin committed:
Implement Level 8 - Dates and Times

misterpuffin committed:
Merge branch 'branch-Level-8'

misterpuffin committed:
Remove unused imports

misterpuffin committed:
Make code more OOP

misterpuffin committed:
Refactor packages

Updated testing script for new folder structure

misterpuffin committed:
Add JUnit tests

misterpuffin committed:
Change to follow Java Coding Standards

misterpuffin committed:
Implement Level 9 - Find

misterpuffin committed:
Remove unused imports

misterpuffin committed:
Add JavaDoc to public methods

marcuspang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+    private final static Tasklist list = new Tasklist();
+
+    public static void main(String[] args) {
+
```
Unnecessary empty line here

marcuspang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+        }
+        ui.customPrint("Bye. Hope to see you again soon!");
+    }
+
+
+
```
There are a few more instances of empty line, so I won't repeat the comment. I think an IDE will resolve this easily

marcuspang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+
+
+
+    private static void handleCommand(Service s, String str) throws DukeException {
```
Missing javadoc here and some of the classes/methods in other files. But I think you will resolve them with the `A-JavaDoc` level

marcuspang commented on src/main/java/duke/data/FileStorage.java:
```diff
@@ -0,0 +1,69 @@
+
+public class FileStorage implements IStorage {
+    private final String filePath;
+    private static final Pattern TASK_REGEX = Pattern.compile("^\\[(T|D|E)\\]\\[(x| )\\] (.*)(?: \\(.*: (.*)\\))?$");
```
I like how you abstracted the regex into a constant to easily retrieve the task data in your `load` function 👍

marcuspang commented on src/main/java/duke/data/FileStorage.java:
```diff
@@ -0,0 +1,69 @@
+
+        try {
+            FileWriter myWriter = new FileWriter(this.filePath);
+            for (Task task: tasklist) {
```
```suggestion
            for (Task task : tasklist) {
```
I feel there should be an empty space in front of the colon here, but it depends on your IDE config. Feel free to close this if you disagree

marcuspang commented on src/main/java/duke/entities/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/[uuuu][uu] HHmm");
+    private static final DateTimeFormatter printFormatter = DateTimeFormatter.ofPattern(("d/M/yyyy HHmm"));
```
>  Constant names must be all uppercase using underscore to separate words.

So perhaps they could be named `DATE_TIME_FORMATTER` and `PRINT_DATE_TIME_FORMATTER`

marcuspang commented on src/main/java/duke/entities/Event.java:
```diff
@@ -0,0 +1,18 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task{
```
```suggestion
public class Event extends Task {
```
Missing a white space here

marcuspang commented on src/main/java/duke/handlers/HandlerFactory.java:
```diff
@@ -0,0 +1,73 @@
+    public IHandler build() {
+        try {
+            Command commandEnum = Command.valueOf(this.command.toUpperCase());
+            switch (commandEnum) {
```
I like how you abstracted the commands into an enum, improves the readability of your code greatly! 👍

marcuspang commented on src/main/java/duke/handlers/HandlerFactory.java:
```diff
@@ -0,0 +1,73 @@
+
+
+public class HandlerFactory {
+    String command;
+    String taskName;
+    String flag;
+    String flagOption;
```
Since you created setter functions for these, these variables should be `private`

marcuspang commented on src/main/java/duke/service/Service.java:
```diff
@@ -0,0 +1,35 @@
+
+    public void addToList(Task t) {
+        this.list.add(t);
+        int size;
+        if ((size = this.list.size()) == 1) {
```
```suggestion
        int size = this.list.size();
        if (size == 1) {

```
Not a coding violation here, just wondering if this logic could be simplified

marcuspang reviewed (commented):
Overall, just some minor coding violations, which would be resolved once you reach the `A-JavaDoc` level. I like how you abstracted the logic appropriately, which helped with the readability of your code greatly 👍

misterpuffin committed:
Update README.md

misterpuffin committed:
Add Checkstyle

misterpuffin committed:
Add Gui

misterpuffin committed:
Merge branch 'master' of https://github.com/misterpuffin/ip

misterpuffin committed:
Fix build.gradle

misterpuffin committed:
Convert to JavaFX

misterpuffin committed:
Add duke functionality to GUI

misterpuffin committed:
Merge branch 'branch-Level-10'

misterpuffin committed:
Update README

This commit expands on the existing explanation in the README to give users a better understanding of the usage and conventions employed in the project

This seeks to improve future code quality by unifying coding standards and reducing the amount of time needed for new developers to onboard the project.

misterpuffin committed:
Add Assertions

misterpuffin committed:
Improve Code quality

misterpuffin committed:
Add streams usage

misterpuffin committed:
Merge pull request #2 from misterpuffin/branch-A-Assertions

Add Assertions

misterpuffin committed:
Merge branch 'master' of https://github.com/misterpuffin/ip into branch-A-Streams

misterpuffin committed:
Merge pull request #3 from misterpuffin/branch-A-Streams

Add streams

misterpuffin committed:
Merge branch 'master' of https://github.com/misterpuffin/ip into branch-A-CodeQuality

misterpuffin committed:
Merge pull request #4 from misterpuffin/branch-A-CodeQuality

Impro vecode quality

misterpuffin committed:
Add github CI

misterpuffin committed:
Implement Undo

misterpuffin committed:
Fix CI problems

misterpuffin committed:
Merge pull request #5 from misterpuffin/branch-C-Undo

Implement Undo

misterpuffin committed:
Improve GUI

misterpuffin committed:
Improve ErrorHandling

misterpuffin committed:
Change Duke to Zupey

misterpuffin committed:
Add Ui.png

misterpuffin committed:
Add User Guide

misterpuffin committed:
Update README.md
